File: 2025-10/2025-10-08/goldilocks_zone.py
def goldilocks_zone(mass):
    luminosity = mass ** 3.5
    goldilocks_zone_start = 0.95 * luminosity ** 0.5
    goldilocks_zone_end = 1.37 * luminosity ** 0.5

    return [round(goldilocks_zone_start, 2), round(goldilocks_zone_end, 2)]

print(goldilocks_zone(1))
print(goldilocks_zone(0.5))
print(goldilocks_zone(6))
print(goldilocks_zone(3.7))
print(goldilocks_zone(20))

File: 2025-10/2025-10-08/goldilocksZone.js
function goldilocksZone(mass) {
  const luminosity = mass ** 3.5;
  const goldilocksZoneStart = 0.95 * Math.sqrt(luminosity);
  const goldilocksZoneEnd = 1.37 * Math.sqrt(luminosity);

  return [+parseFloat(goldilocksZoneStart).toFixed(2), Math.round(goldilocksZoneEnd * 100) / 100];
}

console.log(goldilocksZone(1));
console.log(goldilocksZone(0.5));
console.log(goldilocksZone(6));
console.log(goldilocksZone(3.7));
console.log(goldilocksZone(20));